be/src/formats/parquet/column_reader.cpp

```diff
@@ -188,7 +188,6 @@ void ColumnReader::get_subfield_pos_with_pruned_type(const ParquetField& field,
         if (iceberg_it == subfield_name_2_field_schema.end()) {
             // This suitition should not be happened, means table's struct subfield not existed in iceberg schema
             // Below code is defensive
-            DCHECK(false) << "Struct subfield name: " + format_subfield_name + " not found in iceberg schema.";
```

Comment:
What's the situation the code will go here?

TIcebergSchemaField is the latest schema, and the column's type is original from the newest schema, so here is unreachable.

Reply:
I think this is trying to check when an Iceberg struct evolution has happened but we are scanning a parquet file from before the evolution where the file doesn't contain the newly added subfield column. In that case, btw the comment that this "shouldn't happen" isn't really correct, it's quite normal that it would happen.

Reply:
I disagree with you, must be another problem.

Iceberg is snapshot isolation. Even if the schema is changing, we will only read files based on the latest completed snapshot.

Reply:
Correct, of course. That's referring to the iceberg schema, this is resolving a local file's Parquet schema with the current global table iceberg schema.

Reply:
OK on reading the code, it seems yeah this is already checked by `IcebergMetaHelper::prepare_read_columns`, I can remove this line change.